main.py

```diff
@@ -75,28 +75,43 @@ def instant_code() -> None:
         print('Token could not be verified – did you enter the right SMS PIN?')
 
 
-def impf_me(location: dict):
+def impf_me(location: dict, browser: Browser = None):
     """ Helper function to support concurrency """
-    global b  # Open Browser Helper variable
-    x = b or Browser(**location)
-
     # Keep Browser open
-    if settings.KEEP_BROWSER and not (settings.CONCURRENT_ENABLED):
+    if settings.KEEP_BROWSER:
+      if settings.CONCURRENT_ENABLED:
+        x = browser or Browser(**location)
+
+        if browser is None:
+            logger.info('Keeping Browser open; KEEP_BROWSER is set to True')
+            x.keep_browser = True
+        else:
+            x.reinit(**location)
+      else:
+        global b  # Open Browser Helper variable
+        x = b or Browser(**location)
+
         if b is None:
             logger.info('Keeping Browser open; KEEP_BROWSER is set to True')
             x.keep_browser = True
             b = x
         else:
             x.reinit(**location)
+    else:
+      x = Browser(**location)
```

Comment:
I usually try to avoid if-else statements as it makes for cleaner code.  
If an if-else statement is required, it's usually better to have the shorter instruction come first – so `if not settings.CONCURRENT_ENABLED`